tsdb/show_measurements.go

```diff
@@ -26,7 +27,10 @@ func NewShowMeasurementsExecutor(stmt *influxql.ShowMeasurementsStatement, mappe
 }
 
 // Execute begins execution of the query and returns a channel to receive rows.
-func (e *ShowMeasurementsExecutor) Execute() <-chan *models.Row {
+func (e *ShowMeasurementsExecutor) Execute(closing <-chan struct{}) <-chan *models.Row {
+	// It's important that all resources are released when execution completes.
+	e.close()
```

Comment:
I think `e.close()` could not be invoked here.
Please ref #4926 #4927 
Thanks! @corylanou